From f00fbdfed54dd514b0ed4c2fbc01183d56bd5306 Mon Sep 17 00:00:00 2001
From: XNOR.ai <147757538+xnorai@users.noreply.github.com>
Date: Fri, 15 Nov 2024 16:49:18 -0800
Subject: [PATCH] Open files in readonly mode

---
 gguflib.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gguflib.c b/gguflib.c
index aa68527..73a7b7d 100644
--- a/gguflib.c
+++ b/gguflib.c
@@ -111,7 +111,7 @@ uint64_t gguf_value_len(uint32_t type, union gguf_value *val) {
 
 /* Open a GGUF file and return a parsing context. */
 gguf_ctx *gguf_open(const char *filename) {
-    int fd = open(filename,O_RDWR|O_APPEND);
+    int fd = open(filename,O_RDONLY);
     if (fd == -1) return NULL;
 
     /* Mapping successful. We can create our context object. */
@@ -154,7 +154,7 @@ int gguf_remap(gguf_ctx *ctx) {
     /* Get the size of the file to map, then map it. */
     if (fstat(ctx->fd,&sb) == -1) return 0;
 
-    void *mapped = mmap(0,sb.st_size,PROT_READ|PROT_WRITE,MAP_SHARED,ctx->fd,0);
+    void *mapped = mmap(0,sb.st_size,PROT_READ,MAP_SHARED,ctx->fd,0);
     if (mapped == MAP_FAILED) return 0;
 
     /* Minimal sanity check... */
-- 
2.39.5 (Apple Git-154)

